### PR TITLE
#9 Bhavik/add file upload

### PR DESCRIPTION
**Pull request by br0wncrust4cean**
Added the uploader. 
- Allows you to upload a file, saves to the DB, and then I just display it on the same page for now.

### CODE REVIEW

File: Project/src/main/java/app/FileRepository.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package app;
+
+import app.models.File;
+import app.models.User;
+import org.springframework.data.repository.CrudRepository;
+
+public interface FileRepository extends CrudRepository<File, Long> {
+    File findByUser(User user);
+
```
Comment (DanGravel) [inline Project/src/main/java/app/FileRepository.java]: whitespace

File: Project/target/classes/templates/upload.html [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<!DOCTYPE html>
```
Comment (br0wncrust4cean) [inline Project/target/classes/templates/upload.html]: I need to get rid of this file so ignore for now

File: Project/src/main/java/app/FileUploadController.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        model.addAttribute("files", repository.findAll());
+        return "upload";
+    }
+
+    @PostMapping("/upload")
+    public String uploadFile(@RequestParam("file") MultipartFile file) throws IOException {
+        Object principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        String username = ((UserDetails)principal).getUsername();
+
+        User user = userRepository.findByUsername(username);
```
Comment (DanGravel) [inline Project/src/main/java/app/FileUploadController.java]: We used this somewhere else, should we abstract it away?

Comment (br0wncrust4cean) [reply Project/src/main/java/app/FileUploadController.java]: I abstracted it, take a look.

**Comment by DanGravel** [comment]
+1

**Review by DanGravel — APPROVED** [review]

**Pull request merged**